### PR TITLE
Upgrade to AWS SDK for Go v2

### LINKED ISSUE #1019: switch to aws-sdk-go-v2 as v1 is going away
Context: https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-aws-sdk-for-go-v1-on-july-31-2025/
Migration guide: https://aws.github.io/aws-sdk-go-v2/docs/migrating/

Basically maintenance mode on July 31, 2024 and reach end-of-support on July 31, 2025. The SDK will not 
receive API updates for new or existing services, or be updated to support new regions.

### PR DESCRIPTION
**Pull request by Pluies**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Bump AWS SDK for Go v2 as v1 is deprecated and soon to be unmaintained.

#### Which issue(s) this PR fixes:

Fixes #1019

#### Special notes for your reviewer:

See inline comment on instantiation of the aws client

#### Does this PR introduce a user-facing change?

No

### CODE REVIEW

File: upstream/ami.go [hunk 1]
```diff
@@ -50,13 +55,14 @@ type AMI struct {
...
-func NewAWSClient() *ec2.EC2 {
+func NewAWSClient() *ec2.Client {
 	// Create a new session based on shared / env credentials
-	s := session.Must(session.NewSessionWithOptions(session.Options{
-		SharedConfigState: session.SharedConfigEnable,
-	}))
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		log.Fatal("failed to load aws config", err)
+	}
```
Comment (Pluies) [inline upstream/ami.go:63]: I believe the v2 behaviour of `config.LoadDefaultConfig` is identical to our previous setting of explicitly setting `session.SharedConfigEnable`, as in it'll look into ~/.aws/config and ~/.aws/credentials. At least at I can confirm it works on my machine, correctly picking up config from ~/.aws/config (grabbing creds via a `credential_process` so not the most straightforward config!).

File: go.mod [hunk 2]
```diff
@@ -18,6 +18,20 @@ require (
...
+	github.com/aws/aws-sdk-go-v2/internal/ini v1.8.1 // indirect
+	github.com/aws/aws-sdk-go-v2/service/internal/accept-encoding v1.11.4 // indirect
+	github.com/aws/aws-sdk-go-v2/service/internal/presigned-url v1.11.18 // indirect
+	github.com/aws/aws-sdk-go-v2/service/sso v1.22.5 // indirect
+	github.com/aws/aws-sdk-go-v2/service/ssooidc v1.26.5 // indirect
+	github.com/aws/aws-sdk-go-v2/service/sts v1.30.4 // indirect
+	github.com/aws/smithy-go v1.20.4 // indirect
+)
+
 require (
```
Comment (saschagrunert) [inline go.mod:35]: Can we merge those `require` blocks?

**Review by saschagrunert — COMMENTED** [review]
Just one nit, otherwise LGTM

File: go.mod (hunk 2 above)
Comment (Pluies) [reply go.mod:35]: Totally, on it 👍

Comment (Pluies) [reply go.mod:35]: Updated with direct dependencies in the first block and indirect deps in the second block to match the current setup 👍

**Pull request merged**